src/main/java/com/google/codeu/servlets/UserServlet.java

```diff
@@ -0,0 +1,86 @@
+package com.google.codeu.servlets;
+
+
+import com.google.codeu.data.Datastore;
+import com.google.codeu.data.Message;
+import com.google.codeu.data.User;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+
+@WebServlet("users/*")
+public class UserServlet extends HttpServlet {
+
+    private Datastore datastore;
+
+    @Override
+    public void init() throws ServletException {
+        super.init();
+        datastore = new Datastore();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws IOException, ServletException {
+
+        String requestUrl = request.getRequestURI();
+
+        String user = requestUrl.substring("/users/".length());
+
+        // Confirm that user is valid
+        Set<String> users = datastore.getUsers();
```

Comment:
There is a function getUser here: https://github.com/codeu-spring-2019-team-46/codeu-starter-project/blob/master/src/main/java/com/google/codeu/data/Datastore.java#L58

should we use this instead?

Also this is O(N), since you will need to fetch all users.

Reply:
Oh thanks, I will fix it